### PR TITLE
Speed up checking for NaN for floats

### PR DESCRIPTION
**Pull request by vnlitvinov**
- [x] closes N/A
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This isn't giving much speedup because this is simple change, but for some certain inputs like empty datetime fields in csv it gives some speed (because empty fields are parsed as float `NaN`-s).

### CODE REVIEW

**Comment by vnlitvin** [comment]
Here's what `asv continuous -f 1.01 upstream/master float-nat-speedup -e -b io.csv -a sample_time=1 -a warmup_time=1` shows:

before|after|ratio|test name
-|-|-|-
[c7c4c943]|[54533f3b]||
**master**|**float-nat-speedup**||
1.91±0.03ms|1.86±0.01ms|0.97|io.csv.ReadCSVParseDates.time_multiple_date

I'm run more thorough benchmark now.

File: pandas/_libs/tslibs/nattype.pyx [hunk 1]
```diff
@@ -713,7 +713,14 @@ NaT = c_NaT        # Python-visible
 
 cdef inline bint checknull_with_nat(object val):
     """ utility to check if a value is a nat or not """
-    return val is None or util.is_nan(val) or val is c_NaT
+    cdef:
```
Comment (jreback) [inline pandas/_libs/tslibs/nattype.pyx]: can you move this change to util.is_nan?

Comment (vnlitvin) [reply pandas/_libs/tslibs/nattype.pyx]: Sure, will do.

**Comment by jreback** [comment]
can you run the benchmarks for missing value checking and see what the change is.

**Comment by vnlitvin** [comment]
Sure, I am already running those, but it takes _huuuuge_ time to run when you set warmup and sampling times high enough (and with default settings the results are too flaky to be believable).

**Comment by jreback** [comment]
> Sure, I am already running those, but it takes huuuuge time to run when you set warmup and sampling times high enough (and with default settings the results are too flaky to be believable).

right, though just the missing ones shouldn't be that huge

**Comment by vnlitvin** [comment]
Can you point out these "missing" benchmark names?

**Comment by jreback** [comment]
you can pass a regex to select a subset

**Comment by vnlitvin** [comment]
Could you please recommend what benchmark names might be relevant? I didn't study the whole list of them yet...

**Comment by vnlitvin** [comment]
So running this `asv continuous -f 1.01 upstream/master float-nat-speedup -b '.*missing.*' -b io.csv -e -a sample_time=2 -a warmup_time=2` produces:

before|after|ratio|test name
-|-|-|-
[c7c4c943]|[6e77c81f]||
**master**|**float-nat-speedup**||
1.48±0.01μs|1.40±0.03μs|0.95|timedelta.TimedeltaConstructor.time_from_missing
8.41±0.09ms|7.79±0.06ms|0.93|io.csv.ReadCSVSkipRows.time_skipprows(10000)

**Comment by jreback** [comment]
thanks @vnlitvin

**Pull request merged**